### PR TITLE
Migrate doku - Plugin upgrade guid

### PR DESCRIPTION
**Pull request by swDennis**

### CODE REVIEW

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 1]
```diff
@@ -13,83 +13,129 @@ menu_order: 100
 
 ## Introduction
 
-In this guide we provide you with all essential information you need to keep your plugins Shopware 5.0 compatible.
+In this guide we provide you with all essential information you need to keep your plugins Shopware 5.3 compatible.
 
-Most changes are optional, as the old syntax will still work in most cases.
+## Migrate the Bootstrap.php
+For migrate the bootstrap.php we create a new plugin base file which has the technical name of the plugin.
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: To migrate ...

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 2]
```diff
@@ -13,83 +13,129 @@ menu_order: 100
...
+    }
+}
+```
 
 ## Template extensions
 To ensure your templates files are extensible, neither __extendsTemplate__ nor __extendsBlock__ methods should be used for responsive template. Instead, you should use Shopware's auto loading mechanism.
-The following example shows how template extension plugins need to be updated to achieve the best possible result for Shopware 5 templates.
-The following source code was taken from the SwagExample1 plugin for Shopware 4, which displays a top seller slider and a banner in the article detail page:
+The following example shows how template extension plugins need to be updated to achieve the best possible result for Shopware 5.3 templates.
+The following source code displays a top seller slider and a banner in the article detail page:
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: ... in the product detail page:

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 3]
```diff
@@ -13,83 +13,129 @@ menu_order: 100
...
+            <argument type="service" id="template"/>
+            <argument>%swag_update_plugin.plugin_dir%</argument>
+            <tag name="shopware.event_subscriber"/>
+        </service>
 
+    </services>
+</container>
 ```
+### 2. Create the template
+The goal is to make this plugin compatible with the new Shopware 5 templates. For this purpose, the following should be considered:
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: ... with the Shopware 5 templates ...

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 4]
```diff
@@ -13,83 +13,129 @@ menu_order: 100
...
 
-The goal is to make this plugin compatible with both the new Shopware 5 and the old Shopware 4 templates. For this purpose, the following should be considered:
+    Inside the PostDispatch event, we have  load the corresponding template files.
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: ... , we have to load the ...

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 5]
```diff
@@ -13,83 +13,129 @@ menu_order: 100
...
-- The __extendsTemplate__ function should not be used in the new template, otherwise the plugin template cannot be overwritten by other templates.
-- In order for the plugin template to be easily extended by others, the template adjustments should be extracted into a separate file.
+    In order for the plugin template to be easily extended by others, the template adjustments should be extracted into a separate file.
 
 First, the template structure is revised. The example1.tpl file is now divided into three new files:
 
-- SwagExample1/Views/emotion/detail/example1.tpl (Entry point to extends the emotion template)
-- SwagExample1/Views/responsive/frontend/detail/index.tpl (Entry point to extends the responsive template)
-- SwagExample1/Views/common/frontend/swag_example1/detail_extension.tpl (Contains the source code for the extension)
+    *SwagExample1/Views/emotion/detail/example1.tpl (Entry point to extends the emotion template)
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: the emotion template does not exists anymore, so this point can be deleted

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 6]
```diff
@@ -13,83 +13,129 @@ menu_order: 100
...
+    *SwagExample1/Views/responsive/frontend/detail/index.tpl (Entry point to extends the responsive template)
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: the path is wrong. 

*SwagExample1/Resources/views/frontend/detail/index.tpl (Entry point to extends the template)

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 7]
```diff
@@ -13,83 +13,129 @@ menu_order: 100
...
+    *SwagExample1/Views/common/frontend/swag_example1/detail_extension.tpl (Contains the source code for the extension)
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: this path is also wrong. we do not use a common directory

write the directory you use below: frontend/swag_update_plugin/detail/example1.tpl

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 8]
```diff
@@ -13,83 +13,129 @@ menu_order: 100
...
 
 The new files contain the following source code:
-#### SwagExample1/Views/common/frontend/swag_example1/detail_extension.tpl
+`/.../SwagUpdatePlugin/Resources/views/frontend/detail/index.tpl`
+```smarty
+{extends file="parent:frontend/detail/index.tpl"}
+
+{block name="frontend_detail_index_detail"}
+    {include file="frontend/swag_update_pluigin/detail/example1.tpl"}
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: typo in swag_update_plugin

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 9]
```diff
@@ -110,163 +156,166 @@ The new files contain the following source code:
...
-{block name="frontend_detail_index_detail"}
-    {include file="frontend/swag_example1/detail_extension.tpl"}
-{/block}
-```
+Notice: Template extensions for the responsive template are loaded via the inheritance hierarchy based on the file system. Therefore, this template should be extends via {extends file = ".."}.
 
-#### SwagExample1/Views/responsive/frontend/detail/index.tpl
-```html
-{extends file="parent:frontend/detail/index.tpl"}
+The SwagExample1/Views/responsive/frontend/detail/index.tpl and SwagExample1/Views/emotion/frontend/detail/example1.tpl files serve only as entry points into the original template. The source code for displaying the top seller sliders and the banner element, which was previously located directly in the extended template file, has now been made available globally in a separate template file, and is now simply included by both templates. This has the following advantages:
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: fix the pathes and delete the emotion thing

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 10]
```diff
@@ -110,163 +156,166 @@ The new files contain the following source code:
...
+        <element type="mediaselection">
+            <name>mediaSelection</name>
+            <label lang="de">Bildauswahl</label>
+            <label lang="en">MediaSelection</label>
+        </element>
+    </elements>
+
+</config>
+```
+Add a new element of the type `mediaselection`, set the name a label.
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: ... , set the name and the label.

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 11]
```diff
@@ -110,163 +156,166 @@ The new files contain the following source code:
...
-
 ## Uninstall
 
 During the uninstall process, the user can now be prompted which data he wishes to remove.
-Existing __uninstall()__ method should remove all data, and the new __secureUninstall()__ method should only remove non-user related data.
+Existing __uninstall()__ method should remove all data. 
+Use the `UninstallContext` which contains the __keepUserData()__ method. If the return value is `true` you should only remove non-user related data.
 
-##### Bootstrap.php - How to use 'secureUninstall()'
+### How to use 'secureUninstall()'
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: the secureUninstall method does not exist in the new plugin system

File: source/developers-guide/shopware-5-plugin-update-guide/index.md [hunk 12]
```diff
@@ -110,163 +156,166 @@ The new files contain the following source code:
...
 {
-    return true;
+    public function activate(ActivateContext $context)
+    {
+        $context->scheduleClearCache(InstallContext::CACHE_LIST_DEFAULT);
+    }
+    
+    public function uninstall(UninstallContext $context)
+    {
+        if($context->keepUserData()) {
```
Comment (mitelg) [inline source/developers-guide/shopware-5-plugin-update-guide/index.md]: code style

**Comment by mitelg** [comment]
@swDennis please rebase your branch :blush:

File: source/developers-guide/shopware-5-plugin-update-guide/index.md (hunk 1 above)
Comment (swDennis) [reply source/developers-guide/shopware-5-plugin-update-guide/index.md]: D

File: source/developers-guide/shopware-5-plugin-update-guide/index.md (hunk 2 above)
Comment (swDennis) [reply source/developers-guide/shopware-5-plugin-update-guide/index.md]: D

File: source/developers-guide/shopware-5-plugin-update-guide/index.md (hunk 4 above)
Comment (swDennis) [reply source/developers-guide/shopware-5-plugin-update-guide/index.md]: D

File: source/developers-guide/shopware-5-plugin-update-guide/index.md (hunk 5 above)
Comment (swDennis) [reply source/developers-guide/shopware-5-plugin-update-guide/index.md]: F

File: source/developers-guide/shopware-5-plugin-update-guide/index.md (hunk 6 above)
Comment (swDennis) [reply source/developers-guide/shopware-5-plugin-update-guide/index.md]: D

File: source/developers-guide/shopware-5-plugin-update-guide/index.md (hunk 8 above)
Comment (swDennis) [reply source/developers-guide/shopware-5-plugin-update-guide/index.md]: D

File: source/developers-guide/shopware-5-plugin-update-guide/index.md (hunk 9 above)
Comment (swDennis) [reply source/developers-guide/shopware-5-plugin-update-guide/index.md]: D

File: source/developers-guide/shopware-5-plugin-update-guide/index.md (hunk 10 above)
Comment (swDennis) [reply source/developers-guide/shopware-5-plugin-update-guide/index.md]: D

File: source/developers-guide/shopware-5-plugin-update-guide/index.md (hunk 11 above)
Comment (swDennis) [reply source/developers-guide/shopware-5-plugin-update-guide/index.md]: D

File: source/developers-guide/shopware-5-plugin-update-guide/index.md (hunk 12 above)
Comment (swDennis) [reply source/developers-guide/shopware-5-plugin-update-guide/index.md]: D

**Pull request merged**